### PR TITLE
feat: add signMessageWithoutRand method for kaspa wasm

### PR DESCRIPTION
**Pull request by witter-deland**
#586

### CODE REVIEW

**Comment by witter-deland** [comment]
https://github.com/kasware-wallet/extension/pull/5

**Comment by coderofstuff** [comment]
Signing without aux rand is dangerous and can leak keys.

**Comment by apoelstra** [comment]
>Signing without aux rand is dangerous and can leak keys.

This is completely untrue.

**Comment by coderofstuff** [comment]
> > Signing without aux rand is dangerous and can leak keys.
> 
> This is completely untrue.

Can you elaborate? I'm not familiar with the underlying implementation here but I'm mainly concerned with this leading to nonce reuse.

**Comment by elichai** [comment]
Andrew is obviously right :) 
The auxiliary randomness exist only to mitigate specific kinds of power analysis side-channel attacks,
Providing it definitely improves security, but omitting it should not be considered dangerous, as most legacy signature schemes don't provide mitigations against such attacks,

To read more about the relevant discussions that arised in adding this randomness please see: https://github.com/sipa/bips/issues/195

**Comment by coderofstuff** [comment]
> Andrew is obviously right :) The auxiliary randomness exists only to mitigate specific kinds of power analysis side-channel attacks, Providing it definitely improves security, but omitting it should not be considered dangerous, as most legacy signature schemes don't provide mitigations against such attacks,
> 
> To read more about the relevant discussions that arised in adding this randomness please see: [sipa/bips#195](https://github.com/sipa/bips/issues/195)

Thanks for the input @elichai.

@witter-deland I've reopened the PR while I read up more onto this.

**Comment by witter-deland** [comment]
@coderofstuff Thanks
@elichai @apoelstra Thank you both for participating in the discussion

> > Andrew is obviously right :) The auxiliary randomness exists only to mitigate specific kinds of power analysis side-channel attacks, Providing it definitely improves security, but omitting it should not be considered dangerous, as most legacy signature schemes don't provide mitigations against such attacks,
> > To read more about the relevant discussions that arised in adding this randomness please see: [sipa/bips#195](https://github.com/sipa/bips/issues/195)
> 
> Thanks for the input @elichai.
> 
> @witter-deland I've reopened the PR while I read up more onto this.

**Comment by witter-deland** [comment]
Excuse me. What's going on? This API is very straightforward. Are there any issues that are blocking the merge? Thank you.

**Review by coderofstuff — CHANGES_REQUESTED** [review]
Apologies for the delay, we're currently packed with other things.

The change itself looks sound. I just have some comments on code duplication.

**Comment by witter-deland** [comment]
fixed :)
> Apologies for the delay, we're currently packed with other things.
> 
> The change itself looks sound. I just have some comments on code duplication.

File: wallet/core/src/wasm/message.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ const TS_MESSAGE_TYPES: &'static str = r#"
 export interface ISignMessage {
     message: string;
     privateKey: PrivateKey | string;
+    noAuxRand: boolean;
```
Comment (coderofstuff) [inline wallet/core/src/wasm/message.rs:17]: Please make this optional here. This ensure any existing usage when this is added

File: wallet/core/src/wasm/message.rs [hunk 2]
```diff
@@ -30,10 +31,12 @@ pub fn js_sign_message(value: ISignMessage) -> Result<HexString, Error> {
     if let Some(object) = Object::try_from(&value) {
         let private_key = object.cast_into::<PrivateKey>("privateKey")?;
         let raw_msg = object.get_string("message")?;
+        let no_aux_rand = object.get_bool("noAuxRand")?;
```
Comment (coderofstuff) [inline wallet/core/src/wasm/message.rs:34]: If not passed, this should default to `false`

**Review by coderofstuff — CHANGES_REQUESTED** [review]

**Review by coderofstuff — DISMISSED** [review]
LGTM. Tested the scenarios and existing demo still works.

As a follow-up, please update the `wasm/examples/nodejs/javascript/general/message-signing.js` to include a demo of the new scenario you added support for (passing `noAuxRand: true`)

**Comment by witter-deland** [comment]
Sure, done :)
> LGTM. Tested the scenarios and existing demo still works.
> 
> As a follow-up, please update the `wasm/examples/nodejs/javascript/general/message-signing.js` to include a demo of the new scenario you added support for (passing `noAuxRand: true`)

**Review by coderofstuff — APPROVED** [review]
Thanks for that!

**Pull request merged**